fix(formatter): inconsistent union type output between two runs

Dunqing opened the pull request:
This PR is to fix a union type issue that didn't pass idempotency testing, which was regressed by #16205.

leaysgur commented:
### Merge activity

* **Nov 28, 2:43 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 28, 2:43 AM UTC**: `leaysgur` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Dunqing committed:
fix(formatter): inconsistent union type output between two runs (#16222)

This PR is to fix a union type issue that didn't pass idempotency testing, which was regressed by #16205.
